tests/components/prometheus/test_init.py

```diff
@@ -250,23 +515,26 @@ async def test_sensor_without_unit(
     """Test prometheus metrics for sensors without a unit."""
     body = await generate_latest_metrics(client)
 
-    assert (
-        'sensor_state{domain="sensor",'
-        'entity="sensor.trend_gradient",'
-        'friendly_name="Trend Gradient"} 0.002' in body
-    )
+    EntityMetric(
+        metric_name="sensor_state",
+        domain="sensor",
+        friendly_name="Trend Gradient",
+        entity="sensor.trend_gradient",
+    ).withValue(0.002).assert_in_metrics(body)
 
-    assert (
-        'sensor_state{domain="sensor",'
-        'entity="sensor.text",'
-        'friendly_name="Text"} 0' not in body
-    )
+    EntityMetric(
+        metric_name="sensor_state",
+        domain="sensor",
+        friendly_name="Text",
+        entity="sensor.text",
+    ).withValue(0.0).assert_not_in_metrics(body)
```

Comment:
The `assert_not_in_metrics` method only exists on `EntityMetric`, which I think is good - if we don't want a metric to exist, we don't care what value it has.

But it means that the value passed here is unused, and not needed.

```suggestion
    ).assert_not_in_metrics(body)
```

Reply:
I know this might be silly. Should I add a simple test for each of the `assert_in_metrics()` and `assert_not_in_metrics()` helpers? I added some simple tests for everything else.

Reply:
Adding a test would definitely be good, and help prevent possible regressions when these helper classes are changed in the future.

Reply:
I added tests for the helpers for both classes, both for `assert_in_metrics` and `assert_not_in_metrics`